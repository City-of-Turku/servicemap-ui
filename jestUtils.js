import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });