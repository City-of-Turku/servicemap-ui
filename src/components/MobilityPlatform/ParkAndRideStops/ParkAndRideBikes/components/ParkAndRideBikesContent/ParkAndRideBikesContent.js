import PropTypes from 'prop-types';
import React from 'react';
import { Typography } from '@mui/material';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../../styled/styled';
import TextComponent from '../../../../TextComponent';

const ParkAndRideBikesContent = ({ item }) => {
  const intl = useIntl();
  const itemName = {
    fi: item.name_fi,
    en: item.name_en,
    sv: item.name_sv,
  };

  /**
   * Take string (like kaarina) and return the same string with first character in uppercase.
   * @param {*string} str
   * @returns string
   */
  const toSentenceCase = str => {
    if (str) {
      return str.charAt(0).toUpperCase() + str.slice(1);
    }
    return '';
  };

  const itemAddress = {
    fi: `${item.address_fi}, ${item.address_zip} ${toSentenceCase(item.municipality)}`,
    en: `${item.address_en}, ${item.address_zip} ${toSentenceCase(item.municipality)}`,
    sv: `${item.address_sv}, ${item.address_zip} ${toSentenceCase(item.municipality)}`,
  };

  const parkAndRideInfo = (
    <StyledContainer>
      <StyledHeaderContainer>
        <TextComponent textObj={itemName} isTitle />
      </StyledHeaderContainer>
      <div>
        <StyledTextContainer>
          <Typography variant="body2" component="p">
            {intl.formatMessage({ id: 'mobilityPlatform.parkAndRide.content.subtitle' })}
          </Typography>
        </StyledTextContainer>
        {item.address_fi !== '' ? (
          <TextComponent messageId="mobilityPlatform.content.address" textObj={itemAddress} />
        ) : null}
      </div>
    </StyledContainer>
  );

  return <StyledContainer>{parkAndRideInfo}</StyledContainer>;
};

ParkAndRideBikesContent.propTypes = {
  item: PropTypes.shape({
    name_fi: PropTypes.string,
    name_en: PropTypes.string,
    name_sv: PropTypes.string,
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    address_zip: PropTypes.string,
    municipality: PropTypes.string,
  }),
};

ParkAndRideBikesContent.defaultProps = {
  item: {},
};

export default ParkAndRideBikesContent;
