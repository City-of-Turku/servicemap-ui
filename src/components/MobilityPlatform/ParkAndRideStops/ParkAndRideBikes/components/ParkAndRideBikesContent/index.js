import ParkAndRideBikesContent from './ParkAndRideBikesContent';

export default ParkAndRideBikesContent;
