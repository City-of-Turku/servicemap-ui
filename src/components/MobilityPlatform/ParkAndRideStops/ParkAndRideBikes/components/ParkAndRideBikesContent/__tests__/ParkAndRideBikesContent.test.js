/* eslint-disable react/jsx-props-no-spreading */
// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import { initialState } from '../../../../../../../redux/reducers/user';
import ParkAndRideBikesContent from '../index';
import finnishTranslations from '../../../../../../../i18n/fi';

const mockProps = {
  item: {
    name_fi: 'Testinimi',
    name_en: 'Test name',
    name_sv: 'Test namn',
    address_fi: 'Testiosoite',
    address_en: 'Test address',
    address_sv: 'Test adress',
    address_zip: '20200',
    municipality: 'turku',
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<ParkAndRideBikesContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ParkAndRideBikesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkAndRideBikesContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual('Testinimi');
    expect(p[1].textContent).toContain(finnishTranslations['mobilityPlatform.parkAndRide.content.subtitle']);
    expect(p[2].textContent).toEqual('Osoite: Testiosoite, 20200 Turku');
  });
});
