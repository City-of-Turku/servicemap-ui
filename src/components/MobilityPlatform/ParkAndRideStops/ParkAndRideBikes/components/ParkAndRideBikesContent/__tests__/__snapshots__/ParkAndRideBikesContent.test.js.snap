// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkAndRideBikesContent /> should work 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1ynyhby"
    >
      <div
        class="css-1dznbe2"
      >
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
          >
            Testinimi
          </p>
        </div>
      </div>
      <div>
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Liityntäpysäkki pyörille
          </p>
        </div>
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Osoite: Testiosoite, 20200 Turku
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
